crates/core/src/subscription/query.rs

```diff
@@ -137,12 +119,76 @@ pub fn compile_read_only_query(
     }
 
     if !queries.is_empty() {
-        Ok(Query { queries })
+        Ok(queries.into_iter().collect())
     } else {
         Err(SubscriptionError::Empty.into())
     }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Supported {
+    Scan,
+    Semijoin,
+}
+
+pub fn classify(expr: &QueryExpr) -> Option<Supported> {
```

Comment:
This is probably not very robust. I expected to look for IndexJoin, being exactly the kind of join we support. But the examples from the test always yield JoinInner.